### PR TITLE
Added fix to detect EOF error on queries and retry with plain text

### PR DESCRIPTION
**Pull request by jeffreyxie0615**
https://github.com/lxc/lxd/issues/11388

Added fix to detect EOF error on queries and retry HTTP request with plain text encoding.

### CODE REVIEW

File: lxc/query.go [hunk 1]
```diff
@@ -109,7 +109,9 @@ func (c *cmdQuery) Run(cmd *cobra.Command, args []string) error {
 
 		// If not JSON decoding error then fail immediately.
 		if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) {
-			return err
+			if (err.Error() != "EOF") {
```
Comment (stgraber) [inline lxc/query.go:112]: No need for the brackets in Go, this may also want to try `if err != io.EOF {`

**Comment by stgraber** [comment]
@jeffreyxie0615 you need a quick rebase on your branch to get rid of that `Merge` commit and also edit your commit message to include the required `Signed-off-by: Firstname Lastname <email>` line (see CONTRIBUTING.md).

File: lxc/query.go (hunk 1 above)
Comment (stgraber) [reply lxc/query.go:112]:
```diff
diff --git a/lxc/query.go b/lxc/query.go
index 8554e32f2..7e20fa678 100644
--- a/lxc/query.go
+++ b/lxc/query.go
@@ -108,7 +108,7 @@ func (c *cmdQuery) Run(cmd *cobra.Command, args []string) error {
                var jsonUnmarshalTypeError *json.UnmarshalTypeError
 
                // If not JSON decoding error then fail immediately.
-               if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) {
+               if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) && err != io.EOF {
                        return err
                }
 
```

Comment (stgraber) [reply lxc/query.go:112]: That should be a slightly simpler version of your fix.

Comment (jeffreyxie0615) [reply lxc/query.go]: Should be fixed now!

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**